andela/coding-geeks-ah-backend
# #167484585 implemented email verification link approach

#### What does this PR do?
It implements the approach of sending an email verification link to the user upon a successful registration.

#### Description of Task to be done?
1. Adding a helper for sending emails.
2. Using the helper in the user controller to pass required information for sending an email.
3. Modifying the message property to include a message for notifying the user to check their email for the verification link.
4. Adding a route for verifying the email verification link.
5. Changing the status of a user from **inactive** to **active** after verifying their email address.

#### Screenshots
<img width="579" alt="Screen Shot 2019-08-08 at 14 19 30" src="https://user-images.githubusercontent.com/34879554/62702663-a9de1a80-b9e7-11e9-880d-de8f85e5359d.png">
<img width="865" alt="Screen Shot 2019-08-10 at 18 08 37" src="https://user-images.githubusercontent.com/34879554/62824159-6b3c9180-bb9a-11e9-9756-cbcceaa694b4.png">
<img width="1121" alt="Screen Shot 2019-08-10 at 18 10 50" src="https://user-images.githubusercontent.com/34879554/62824162-71cb0900-bb9a-11e9-8ef7-9178033e17a7.png">
<img width="1393" alt="Screen Shot 2019-08-10 at 18 31 10" src="https://user-images.githubusercontent.com/34879554/62824381-0fbfd300-bb9d-11e9-8e19-7dc950ca98ac.png">

#### How should this be manually tested?
1. Create an account on sendgrid.com
2. Create an API key for sending emails on **www.sendgrid.com**.
3. Update your .env file with the following: SENDGRID_API_KEY=**putthekeyhere**, EMAIL_SENDER=no-reply@authorshaven.com, APP_URL=localhost.
1. Clone this branch.
2. Do **npm install**.
3. Register a new user and after successful registration, Check your email for a verification link. If you do not get it your inbox, check your spam folder. It should be there!
4. Copy the link and paste it in Postman and check for the verification confirmation link.
